pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeFieldSpec.java

```diff
@@ -62,6 +62,21 @@ public FieldType getFieldType() {
     return FieldType.TIME;
   }
 
+  @Override
+  public void setName(String name) {
+    // Ignore setName for TimeFieldSpec because we pick the name from TimeGranularitySpec.
```

Comment:
No. Because we cannot control the order of all setters when using ObjectMapper, we cannot tell a mismatch when setting the name. This method is automatically called by the ObjectMapper, so treat it as a no-op should be sufficient.